binderhub/tests/test_ratelimit.py

```diff
@@ -0,0 +1,81 @@
+import time
+from unittest import mock
+
+import pytest
+
+from binderhub.ratelimit import RateLimiter, RateLimitExceeded
+
+
+def test_rate_limit():
+    r = RateLimiter(limit=10, period_seconds=60)
+    assert r._limits == {}
+    now = r.time()
+    reset = int(now) + 60
+    with mock.patch.object(r, "time", lambda: now):
+        limit = r.increment("1.2.3.4")
+    assert limit == {
+        "remaining": 9,
+        "reset": reset,
+    }
+    assert r._limits == {
+        "1.2.3.4": limit,
+    }
+    for i in range(1, 9):
```

Comment:
`(0, 9)` or `(1, 10)`?: `r.increment` is called once prior to this loop, so you should be able to call it a further 9 times without an exception.

Reply:
```suggestion
    for i in range(1, 10):
```
so we can keep the `assert` about the remaining requests the same below